### PR TITLE
#162414271 Integrate social authentication

### PR DESCRIPTION
**Pull request by jesseinit**
#### What does this PR do?

- Implement authentication via social networks

#### Description of Task to be completed?

- Write tests
- Install passport strategies
- Add implementation to route file
- Update tests

#### How should this be manually tested?

- Clone the repo `git clone https://github.com/andela/neon-ah-backend.git`
- Install Postgresql from [here](https://www.postgresql.org/download/)
- Create a database `haven_dev`
- Checkout to the branch `git checkout -b ft-integrate-social-auth-162414271`
- Install application dependencies `npm install`
- Run the tests `npm test`

#### What are the relevant pivotal tracker stories?

[Social Authentication](https://www.pivotaltracker.com/story/show/162414271)

### CODE REVIEW

File: server/controllers/UserController.js [hunk 1]
```diff
@@ -71,24 +75,21 @@ class UserController {
...
 
       if (!isPasswordEqual) {
-        return res.status(400).send({
+        return res.status(400).json({
           status: 'failure',
-          message: 'Password does not match'
+          data: {
+            statusCode: '404',
+            message: 'Password does not match',
+            error: 'Password does not match'
```
Comment (Bobsar0) [inline server/controllers/UserController.js]: Error and message fields have duplicate values. Why not merge into say message?

File: server/controllers/UserController.js [hunk 2]
```diff
@@ -99,37 +98,118 @@ class UserController {
...
+          message: 'Password has been successfully updated. Kindly login.'
+        }
       });
     } catch (error) {
       if (error.name === 'TokenExpiredError' || error.name === 'JsonWebTokenError') {
         return res.status(401).send({
           status: 'failure',
-          message: error.name
+          data: {
+            statusCode: '200',
```
Comment (tejiri4) [inline server/controllers/UserController.js]: please check your status code. there is error 401 and 200.

**Comment by tejiri4** [comment]
Excellent, Well done Jesse LGTM

File: server/controllers/UserController.js (hunk 2 above)
Comment (jesseinit) [reply server/controllers/UserController.js]: Thanks for your observation. Lemme fix it

**Review by rafmme — COMMENTED** [review]
LGTM, a job well done.

**Review by seunkoko — APPROVED** [review]

**Pull request merged**